MetaMask/metamask-extension
# Replace SiteIcon with AvatarFavicon in PermissionsRedirect

## **Description**

Fixed #20459 

Implemented AvatarFavicon to replace SiteIcon components on the PermissionsRedirect page, marking the final instance of its usage. Consequently, the SiteIcon component files have been removed, along with their exclusion from the ts-migration-dashboard list. This streamlines the codebase and aligns it with current component standards.

Note: SiteIcon previously set to 64px; AvatarFaviconSize is set to Xl (48px).

## **Related issues**
1. Part of #17670 
2. Part of #19526 

## **Manual testing steps**

1. Go to [e2e Test Dapp](https://metamask.github.io/test-dapp/)
2. Connect MM to Dapp
a) Approve request for connection
b) Observe icons are correctly rendered on `Connecting` page before redirect
c) Observe all other text and UI elements are unchanged.

## **Screenshots/Recordings**



### **Before**


![permissions-redirect_before](https://github.com/MetaMask/metamask-extension/assets/58456495/9dd2ddb6-7465-4781-bc21-79f25f76b43c)
Recordings: [permissions-redirect_before.webm](https://github.com/MetaMask/metamask-extension/assets/58456495/0e465686-ad90-45da-aeec-85de62b59505)


### **After**


![permissions-redirect_after](https://github.com/MetaMask/metamask-extension/assets/58456495/f1042dcd-420a-4709-b859-3324e6322e5b)
Recordings: [permissions-redirect_after.webm](https://github.com/MetaMask/metamask-extension/assets/58456495/4a59e93a-83d5-45e0-99bf-41fedb31b887)


## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
